src/test/scala/com/campudus/tableaux/TableauxTestBase.scala

```diff
@@ -49,26 +49,40 @@ trait TableauxTestBase extends TestVerticle {
     }
   }
 
+  def exceptionTest(f: => Future[_]): Unit = {
+    (try f catch {
+      case ex: Throwable => Future.failed(ex)
```

Comment:
This would let the test succeed in a case where `f` doesn't even get a future back because of an exception. I doubt that you'd want that... You always want a reply back - you may get a failed future, but the methods shouldn't throw exceptions. I think you can just get rid of the `try ... catch` path and it would still work. Or have a `fail` in here, that tells you that it shouldn't fail with an exception but a failed future :)